qa/tasks/deepsea.py

```diff
@@ -363,8 +404,64 @@ def _introspect_roles(self, deepsea_ctx):
         for var in report_vars:
             self.log.info("{} == {}".format(var, deepsea_ctx[var]))
 
+    def _master_python_version(self, py_version):
+        """
+        Determine if a given python version is installed on the Salt Master
+        node.
+        """
+        python_binary = 'python{}'.format(py_version)
+        installed = True
+        try:
+            self.master_remote.run(args=[
+                'type',
+                python_binary,
+                run.Raw('>'),
+                '/dev/null',
+                run.Raw('2>&1'),
+                ])
+        except CommandFailedError:
+            installed = False
+        if installed:
+            self.master_remote.run(args=[
+                python_binary,
+                '--version'
+                ])
+        else:
+            self.log.info(
+                '{} not installed on master node'.format(python_binary)
+                )
+        return installed
+
+    def _maybe_apply_alternative_defaults(self):
+        global_yml = '/srv/pillar/ceph/stack/global.yml'
+        if self.alternative_defaults:
+            data = ''
+            for cnf in self.alternative_defaults:
+                for k, v in cnf.items():
+                    info_msg = (
+                        "Applying alternative default {}: {}".format(k, v)
+                        )
+                    data += "{}: {}\n".format(k, v)
+                    self.log.info(info_msg)
+            if data:
+                sudo_append_to_file(
+                    self.master_remote,
+                    global_yml,
+                    data,
+                    )
+        try:
+            self.master_remote.run(
+                args="test -f {}".format(global_yml)
+                )
+        except CommandFailedError:
+            return None
+        self.master_remote.run(
+            args="ls -l {p} ; cat {p}".format(p=global_yml)
```

Comment:
This will not needed if you use 'tee' instead of 'cat'.

Reply:
this code has been reworked in a later PR. Maybe it would make more sense if you did your review in #232 ? That's the final product.

Reply:
> this code has been reworked in a later PR. Maybe it would make more sense if you did your review in #232 ? That's the final product.

I'm sorry I've totally lost in those PR, the github is not good when it comes to have different PRs based on other PRs...